R/license_note.R

```diff
@@ -1,97 +1,120 @@
-      ),
-      class = "rextendr_error"
-    )
+  # exclude current package from LICENSE.note
+  current_package <- metadata[["resolve"]][["root"]]
+
+  packages <- packages[packages[["id"]] != current_package, ]
+
+  replace_na <- function(data, replace = NA, ...) {
+    if (vctrs::vec_any_missing(data)) {
+      missing <- vctrs::vec_detect_missing(data)
+      data <- vctrs::vec_assign(data, missing, replace,
+        x_arg = "data",
+        value_arg = "replace"
+      )
+    }
+    data
   }
 
-  list_license <- processx::run(
-    "cargo",
-    c(
-      "license",
-      "--authors",
-      "--json",
-      "--avoid-build-deps",
-      "--avoid-dev-deps",
-      "--manifest-path", manifest_file
-    )
-  )$stdout %>%
-    jsonlite::parse_json()
-
-  package_names <- processx::run(
-    "cargo",
-    c(
-      "metadata",
-      "--no-deps",
-      "--format-version", "1",
-      "--manifest-path", manifest_file
+  packages[["respository"]] <- replace_na(
+    packages[["repository"]],
+    "unknown"
+  )
+
+  packages[["licenses"]] <- replace_na(
+    packages[["repository"]],
+    "not provided"
+  )
+
+  prep_authors <- function(authors, package) {
+    authors <- ifelse(
+      is.na(authors),
+      paste0(package, " authors"),
+      authors
     )
-  )$stdout %>%
-    jsonlite::parse_json() %>%
-    purrr::pluck("packages") %>%
-    purrr::map_chr("name")
-
-  .prep_authors <- function(authors, package) {
-    ifelse(!is.null(authors), authors, paste0(package, " authors")) %>%
-      stringi::stri_replace_all_regex(r"(\ <.+?>)", "") %>%
-      stringi::stri_replace_all_regex(r"(\|)", ", ")
+
+    authors <- stringi::stri_replace_all_regex(authors, r"(\ <.+?>)", "")
+
+    paste0(authors, collapse = ", ")
   }
 
+  packages[["authors"]] <- unlist(Map(
+    prep_authors,
+    packages[["authors"]],
+    packages[["name"]]
+  ))
+
   separator <- "-------------------------------------------------------------"
 
-  note_header <- paste0(
-    "The binary compiled from the source code of this package contains the following Rust crates:\n",
-    "\n",
-    "\n",
-    separator
+  note_header <- glue::glue(
+    "
+    The binary compiled from the source code of this package \\
+    contains the following Rust crates:
+
+
+    {separator}
+    "
   )
 
-  note_body <- list_license %>%
-    purrr::discard(function(x) x$name %in% package_names) %>%
-    purrr::map_chr(
-      function(x) {
-        paste0(
-          "\n",
-          "Name:        ", x$name, "\n",
-          "Repository:  ", x$repository, "\n",
-          "Authors:     ", .prep_authors(x$authors, x$name), "\n",
-          "License:     ", x$license, "\n",
-          "\n",
-          separator
-        )
-      }
-    )
+  note_body <- glue::glue_data(
+    packages,
+    "
+
+    Name:        {name}
+    Repository:  {repository}
+    Authors:     {authors}
+    License:     {license}
```

Comment:
can we use paste or something similar here that isn't `glue::glue_data()` this is going to make it harder to remove glue at a later point.